server/src/main/java/org/elasticsearch/action/support/replication/TransportReplicationAction.java

```diff
@@ -117,6 +117,20 @@ protected enum SyncGlobalCheckpointAfterOperation {
         AttemptAfterSuccess
     }
 
+    /**
+     * Execution of the replica action
+     */
+    protected enum ReplicaActionExecution {
```

Comment:
Ah could we split this one out too in a separate PR, for similar reasons to #111017? That's going to reduce the noise in this PR massively.